src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceUpdateSettings.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Configuration
+{
+    public class PackageSourceUpdateSettings
+    {
+        public bool UpdateCredentials { get; set; }
+        public bool UpdateEnabled { get; set; }
```

Comment:
How about making these read only (getter only). There's little to no benefit allowing the setters, and it allows APIs that use this class to make safe multi-threading assumptions.  Changing them to add setters in the future will not be a breaking change, but removing setters is.